src/board/system76/common/battery.c

```diff
@@ -112,3 +125,38 @@ void battery_reset(void) {
     battery_start_threshold = BATTERY_START_THRESHOLD;
     battery_end_threshold = BATTERY_END_THRESHOLD;
 }
+
+// Read the charge thresholds from flash. Falls back to defaults if not found.
+bool battery_load_thresholds(void) {
+    bool loaded = true;
+
+    flash_read(BAT_CFG_ADDR, (uint8_t *)&bat_cfg, sizeof(bat_cfg));
+
+    if (bat_cfg.magic != BAT_CFG_MAGIC) {
+        bat_cfg.magic = BAT_CFG_MAGIC;
+        bat_cfg.start_threshold = BATTERY_START_THRESHOLD;
+        bat_cfg.end_threshold = BATTERY_END_THRESHOLD;
+        loaded = false;
+    }
+
+    battery_start_threshold = bat_cfg.start_threshold;
+    battery_end_threshold = bat_cfg.end_threshold;
```

Comment:
These values should probably be tested for validity the same way they are in the ACPI code, by calling the battery_set_end_threshold function first and the battery_set_start_threshold function second.